Add zero constants/methods for each quantity

Add `constant`? Add `const fn`? `impl Zero`?

I don't think it's possible with the way `Quantity` is currently defined. 

```
   ::: src/system.rs:243:9
    |
243 | /         pub struct Quantity<D, U, V>
244 | |         where
245 | |             D: Dimension + ?Sized,
246 | |             U: Units<V> + ?Sized,
...   |
256 | |             pub value: V,
257 | |         }
    | |_________- required by `si::Quantity`
```

Placing `$quantity::new_in_base_units()` below `$quantity::new()` (in the same impl block) hits *"error[E0723]: trait bounds other than `Sized` on const fn parameters are unstable"*. That impl block requires `U: Units<V>` and `V: Num + Conversion<V>`.

Because the struct definition contains a where clause, any function creating a `Quantity` also needs to have the same where clause. Putting `$quantity::new_in_base_units()` in its own impl block gives me 312 instances of the following:

```
error[E0277]: the trait bound `U: si::Units<V>` is not satisfied
   --> src/quantity.rs:290:56
    |
290 |               pub const fn new_in_base_unit(value: V) -> Self {
    |                                                          ^^^^
    |
   ::: src/si/volume_rate.rs:3:1
    |
3   | / quantity! {
4   | |     /// Volume rate (base unit cubic meter per second, m³ · s⁻¹).
5   | |     quantity: VolumeRate; "volume rate";
6   | |     /// Dimension of volume rate, L³T⁻¹ (base unit cubic meter per second, m³ · s⁻¹).
...   |
149 | |     }
150 | | }
    | |_- in this macro invocation
    |
   ::: src/system.rs:243:9
    |
243 | /         pub struct Quantity<D, U, V>
244 | |         where
245 | |             D: Dimension + ?Sized,
246 | |             U: Units<V> + ?Sized,
...   |
256 | |             pub value: V,
257 | |         }
    | |_________- required by `si::Quantity`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `U`
    |
287 |         impl<U: si::Units<V>, V> $quantity<U, V> {
    |               ^^^^^^^^^^^^^^
```

I know the Rust API guidelines [recommend](https://rust-lang.github.io/api-guidelines/future-proofing.html#data-structures-do-not-duplicate-derived-trait-bounds-c-struct-bounds) not putting where clauses on a generic struct's declaration and only placing it on impl blocks, but to change that now would be a breaking change and a making a new major release for a single `const fn` constructor is probably not worth it.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -456,6 +456,20 @@ pub trait ConversionFactor<V>:
     fn value(self) -> V;
 }
 
+/// Helper trait to identify the zero value of a type at compile time.
+///
+#[cfg_attr(all(feature = "si", feature = "f32"), doc = " ```rust")]
+#[cfg_attr(not(all(feature = "si", feature = "f32")), doc = " ```rust,ignore")]
+/// # use uom::si::f32::Length;
+/// use uom::ConstZero;
+///
+/// const ORIGIN: (Length, Length, Length) = (Length::ZERO, Length::ZERO, Length::ZERO);
+/// ```
+pub trait ConstZero {
+    /// Constant representing the zero value.
+    const ZERO: Self;
+}
+
 /// Default [kind][kind] of quantities to allow addition, subtraction, multiplication, division,
 /// remainder, negation, and saturating addition/subtraction.
 ///
@@ -507,6 +521,10 @@ storage_types! {
             self
         }
     }
+
+    impl crate::ConstZero for V {
+        const ZERO: Self = 0.0;
+    }
 }
 
 storage_types! {
@@ -531,6 +549,10 @@ storage_types! {
         fn value(self) -> V {
             self.to_integer()
         }
+    }
+
+    impl crate::ConstZero for V {
+        const ZERO: Self = 0;
     }
 }
 

--- a/src/system.rs
+++ b/src/system.rs
@@ -1288,6 +1288,19 @@ macro_rules! system {
             }
         }
 
+        impl<D, U, V> $crate::ConstZero for Quantity<D, U, V>
+        where
+            D: Dimension + ?Sized,
+            U: Units<V> + ?Sized,
+            V: $crate::num::Num + $crate::Conversion<V> + $crate::ConstZero,
+        {
+            const ZERO: Self = Self {
+                dimension: $crate::lib::marker::PhantomData,
+                units: $crate::lib::marker::PhantomData,
+                value: V::ZERO,
+            };
+        }
+
         serde! {
         impl<D, U, V> $crate::serde::Serialize for Quantity<D, U, V>
         where

